src/client/renderingengine.cpp

```diff
@@ -567,6 +568,7 @@ void RenderingEngine::draw_anaglyph_3d_mode(Camera *camera, bool show_hud, Hud *
 			irr::scene::ESNRP_TRANSPARENT +
 			irr::scene::ESNRP_TRANSPARENT_EFFECT + irr::scene::ESNRP_SHADOW;
 	camera->getCameraNode()->setPosition(rightEye);
```

Comment:
About half of the code in each 3D mode function is replicated too in each function (calculating eye positions, restoring the camera position), certainly could use refactoring.

As a newbie: I guess it would be OK to add that kind of changes under this pull request (judging from the fact that you suggest it), but what about other changes, like changes to the 3D calculations themselves? Where lies the limit of adding to old request and making a new one?